File: backend/app/schemas/resource_investigation.py
from datetime import datetime
from typing import Any
from pydantic import BaseModel
from app.schemas.common import DatetimeStr

class ResourceInvestigationGenerateRequest(BaseModel):
    custom_instruction: str | None = None

class ResourceGap(BaseModel):
    category: str = ""
    needed: str = ""
    reason: str = ""
    severity: str = ""

class ResourceInvestigationSummary(BaseModel):
    internal_resource_count: int = 0
    external_resource_count: int = 0
    internal_by_category: dict[str, int] = {}
    external_by_category: dict[str, int] = {}
    resource_gaps: list[ResourceGap] = []
    key_findings: list[str] = []
    overall_assessment: str = ""

class ResourceInvestigationReportResponse(BaseModel):
    id: str
    enterprise_id: str
    title: str
    content: str
    # 与风险评估报告一致：JSONB summary 原样透传，避免过滤 chapters 等扩展键。
    summary: dict[str, Any]
    status: str
    generated_by: str
    generated_at: DatetimeStr | None
    created_at: DatetimeStr
    updated_at: DatetimeStr
    model_config = {"from_attributes": True}

class ResourceInvestigationPreviewResponse(BaseModel):
    report_id: str
    title: str
    html: str

File: frontend/src/types/resourceInvestigation.ts
export interface ResourceGap {
  category: string;
  needed: string;
  reason: string;
  severity: string;
}

export interface ResourceInvestigationSummary {
  internal_resource_count: number;
  external_resource_count: number;
  internal_by_category: Record<string, number>;
  external_by_category: Record<string, number>;
  resource_gaps: ResourceGap[];
  key_findings: string[];
  overall_assessment: string;
}

export interface ResourceInvestigationReport {
  id: string;
  enterprise_id: string;
  title: string;
  content: string;
  summary: Partial<ResourceInvestigationSummary> & {
    chapters?: Array<{ key: string; title: string; content: string }>;
  };
  status: "draft" | "generating" | "completed";
  generated_by: "ai" | "manual";
  generated_at: string | null;
  current_version?: number;
  created_at: string;
  updated_at: string;
  style_preference?: Record<string, string> | null;
}

export interface ResourceInvestigationPreview {
  report_id: string;
  title: string;
  html: string;
}
